build_tools/azure/install.sh

```diff
@@ -133,19 +133,19 @@ try:
 except ImportError:
     print('pandas not installed')
 "
-python -m pip list
+# Automatically use as many cores as possible to speed-up scikit-learn
+# build.
+pip install loky
+export SKLEARN_BUILD_PARALLEL="auto"
 
+python -m pip list
 if [[ "$DISTRIB" == "conda-pip-latest" ]]; then
-    # Check that pip can automatically install the build dependencies from
-    # pyproject.toml using an isolated build environment:
+    # Check that pip can automatically install missing build dependencies from
+    # pyproject.toml.
```

Comment:
From my understanding, `pip install -e .` is using a isolated build when the `--no-build-isolation` flag is not set.

Reply:
Yes indeed, I did a test to confirm this and it's building in isolated env even with `-e`. I was confused and thought it was not longer the case with recent pip but I was wrong. Let me restore the old comment and keep the isolated build. I think it's a good idea to have at least one job that uses the pyproject.toml build dependencies.